examples/nas/enas/macro.py

```diff
@@ -18,16 +18,16 @@ def __init__(self, key, num_prev_layers, in_filters, out_filters):
             PoolBranch('avg', in_filters, out_filters, 3, 1, 1),
             PoolBranch('max', in_filters, out_filters, 3, 1, 1)
         ])
-        if num_prev_layers > 0:
-            self.skipconnect = mutables.InputChoice(num_prev_layers, n_selected=None, reduction="sum")
+        if len(prev_labels) > 0:
+            self.skipconnect = mutables.InputChoice(choose_from=prev_labels, n_chosen=None, reduction="sum")
```

Comment:
we should think about what if the inputs are not all from layerchoice but from normal layers, then how to specify `choose_from`

Reply:
Normal layers should be wrapped with a mutable scope to gain the power from the keys. Maybe we can do some annotations.